File: app/src/main/java/tourguide/tourguidedemo/MyApplication.java
package tourguide.tourguidedemo;

import android.app.Application;
import android.content.Context;

import com.squareup.leakcanary.LeakCanary;
import com.squareup.leakcanary.RefWatcher;

/**
 * Created by tanjunrong on 7/3/15.
 */
public class MyApplication extends Application {
    private RefWatcher _refWatcher;

    public static RefWatcher getRefWatcher(Context context) {
        MyApplication application = (MyApplication) context.getApplicationContext();
        return application._refWatcher;
    }

    @Override public void onCreate() {
        super.onCreate();
        /* This is for checking memory leak: https://github.com/square/leakcanary */
        _refWatcher = LeakCanary.install(this);
    }
}

File: app/src/main/java/tourguide/tourguidedemo/MyApplication.kt
package tourguide.tourguidedemo

import android.app.Application
import android.content.Context

import com.squareup.leakcanary.LeakCanary
import com.squareup.leakcanary.RefWatcher

class MyApplication : Application() {
    private var _refWatcher: RefWatcher? = null

    override fun onCreate() {
        super.onCreate()
        /* This is for checking memory leak: https://github.com/square/leakcanary */
        _refWatcher = LeakCanary.install(this)
    }

    companion object {

        fun getRefWatcher(context: Context): RefWatcher? {
            val application = context.applicationContext as MyApplication
            return application._refWatcher
        }
    }
}
